README.md

```diff
@@ -40,85 +37,146 @@ SDK library for [Node.js][node] to consume [UnivaPay][univapay-url] API.
+```bash
+# With npm (prefered)
+npm install --save univapay-node
+npm install --save-dev tslib # with JavaScript, needed to use the sdk
 
+# With yarn
+yarn add univapay-node
+yarn add --dev tslib # with JavaScript, needed to use the sdk
+```
 
-Requirements
-------------
+## Usage
 
-`univapay-node` requires the following to run:
+### Requirements
 
-  * [Node.js][node] 10 or 12
-  * [npm][npm] (normally comes with Node.js) or [yarn][yarn]
+If you do not already have your store application token, please create it first:
 
+-   Go to the `Store > For developpers > Application tokens` page
+-   Click on `Add`
+-   Add you domain and your mode
+-   Copy the JWT from the created token
+-   Do not forget to store your secret
 
-Installation
-------------
+### Setup SDK
 
-UnivaPay Node SDK is easiest to use when installed with [npm][npm]:
+```typescript
+import SDK from "univapay-node";
 
-```bash
-npm install univapay-node
-```
-or with [yarn][yarn]:
-```bash
-yarn add univapay-node
-```
+const apiEndpoint = "https://api.univapay.com"; // TODO: Investigate env variable
+const storeJwt = jwt; // see `Requirements`
+const storeJwtSecret = secret; // see `Requirements`
 
-Usage
------
+const sdk = new SDK({
+    endpoint: apiEndpoint,
+    jwt: storeJwt,
+    secret: storeJwtSecret,
+});
+```
 
-Just import the module into your code with, create an instance of it and you're set:
+### Create charge
 
-```javascript
+```typescript
 import SDK from "univapay-node";
+import { PaymentType, TransactionTokenType } from "univapay-node/resources/TransactionTokens";
+import { ResponseError } from "univapay-node/errors/RequestResponseError";
+
+const sdk = new SDK({ endpoint, jwt, secret });
+
+// Create charge token
+let transactionToken;
+try {
+  const transactionToken = await sdk.transactionTokens.create({
+    type: TransactionTokenType.ONE_TIME, // Can be used only once
+    paymentType: PaymentType.CARD,
+
+    // Data type varies depending on the payment type
+    data: { cardholder, cardNumber, expMonth, expYear, cvv }
+  });
+} catch (tokenCreatError: ResponseError) {
+  handleError(errtokenCreatErroror);
+}
 
-const sdk = new SDK();
+// Create charge
+try {
+  const charge = await sdk.charges.create({
+    amount: 1000;
+    currency: "JPY",
+    transactionTokenId: transactionToken.id,
+  });
+} catch (chargeCreateError: ResponseError) {
+  // Cleanup unused token to recreate another one for the next charge
+  await sdk.transactionTokens.delete(transactionToken.id);
+
+  handleError(chargeCreateError);
+}
 ```
 
-### Customize
+### Polling
 
-You can create your own `SDK` object configuration that only contains selected resources. In the following example show
-how to prepare a class that only has `Stores` resources available:
+After creating a charge the status will rarely directly be `successful` or `failed` and can be `pending` until the API fully processes it. You can poll the charge if you need to know if the charge failed or suceeded.
```

Comment:
`rarely` could be subjective, perhaps "After creating a charge, the status will be initialised as `pending`, and eventually be updated to `successful` or `failed`" ?